docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1379,6 +1364,120 @@ As a Zephyr OS user I want to... ???
 
 [/SECTION]
 
+[SECTION]
+TITLE: Thread Synchronization
+
+[REQUIREMENT]
+UID: ZEP-990001
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Counting Semaphore Creation
+STATEMENT: >>>
+The system shall provide a mechanism to create semaphores, both statically at build time and dynamically at runtime, ensuring deterministic operation with bounded execution times.
+<<<
+USER_STORY: >>>
+As a software developer tasked with implementing resource management in our real-time operating system (RTOS), I want to integrate a semaphore synchronization primitive to facilitate coordinated access to shared resources among multiple tasks.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-990002
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore Initialization
+STATEMENT: >>>
+The system shall provide a mechanism to initialize semaphore values to regulate the number of tasks or resources concurrently accessing a shared resource.
```

Comment:
**suggestion:** The RQT provides a rationale in its statement: "... to regulate the number ...." which could be moved to an "informative" field of the RQT. It also might be misleading as there is no strict connection between a semaphore count and the number of threads interacting with said semaphore. 

**nitpick:** A semaphore is a signalling device not neccesarely tied to the coordination of shared resource access